### DISCUSSION TITLE
Adding Kerberos to Airflow Worker Pods (k8s)

### DISCUSSION THREAD

**Discussion by jlouk** [apache/airflow#17948] [General] [answered]
I am currently running airflow in k8s with this helm chart. When I enable kerberos it appears that kerberos is enabled on the scheduler node but not in the pods created for each task of a dag.

I was wondering if it was possible to have the KubernetesExecutor spin up a new pod and run a `kinit ...` prior to running the associated pod task.  Is there something that I could be missing?

If its not possible, what would the risk be of adding a configuration value and maybe some logic inside of airflow/kubernetes/pod_generator.py so that we could have kerberos enabled worker pods?

Thanks and apologies if this is a simple question

**Comment by potiuk** [comment, marked as answer]
Kerberos in the Airlfow Chart works by default only with Celery Executors (and the Celery Workers for CeleryKubernetesExecutor). 

The Kerberos implementation in Helm is based on the assumption that there is a long running worker container and there is a long running kerberos sidecar that performs periodic refresh of the token from kerberos. It makes very little sense to have similar approach for Kubernetes Excutor where the Pod is "One time use only". A

IMHO dding a generic support for K8S executor to refresh kerberos token would make very little sense because this token will be used once anyway and it is much easier if you do it in your custom Pod template.

What you really need to do is to write your own pod template where you'd map the kerberos keytab into and run `airflow kerberos` command to refresh the token and then run your stuff. That means that you simply need to map some of your secret containing keytab into your pod template and have a command that will run the airlfow kerberos command before.


UPDATE. I thought a bit and I think you could even have a similar approach as workers have - where kerberos refresh would run as an init container and store the token in a shared volume and then run your job with that token. This would add extra security (your job would never see the keytab) and actually might be good idea to contribute back eventually - but for now I think it's easier if you still do it via pod template.

**Reply by jlouk** [reply]
Thanks for your feedback I appreciate it!
